src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContactBubbleStab.cpp

```diff
@@ -0,0 +1,1150 @@
+                                                           CRSMatrix< real64, globalIndex > & localMatrix,
+                                                           ParallelVector & rhs,
+                                                           ParallelVector & solution,
+                                                           bool const GEOS_UNUSED_PARAM( setSparsity ) )
+{
+
+
+  // setup monolithic coupled system
+
+  // Create the lists of interface elements that have same type.
+  createFaceTypeList( domain );
+
+  // Create the lists of interface elements that have same type and same fracture state.
+  updateStickSlipList( domain );
+
+  // Create the list of cell elements that they are enriched with bubble functions.
+  createBubbleCellList( domain );
+
+  dofManager.setDomain( domain );
+  setupDofs( domain, dofManager );
+  dofManager.reorderByRank();
+
+  // Set the sparsity pattern without the Abu and Aub blocks.
+  SparsityPattern< globalIndex > patternDiag;
+  dofManager.setSparsityPattern( patternDiag );
+
+  // Get the original row lengths (diagonal blocks only)
+  array1d< localIndex > rowLengths( patternDiag.numRows() );
+  for( localIndex localRow = 0; localRow < patternDiag.numRows(); ++localRow )
+  {
+    rowLengths[localRow] = patternDiag.numNonZeros( localRow );
+  }
+
+  // Add the number of nonzeros induced by coupling
+  this->addCouplingNumNonzeros( domain, dofManager, rowLengths.toView() );
+
+  // Create a new pattern with enough capacity for coupled matrix
+  SparsityPattern< globalIndex > pattern;
+  pattern.resizeFromRowCapacities< parallelHostPolicy >( patternDiag.numRows(), patternDiag.numColumns(), rowLengths.data() );
+
+  // Copy the original nonzeros
+  for( localIndex localRow = 0; localRow < patternDiag.numRows(); ++localRow )
+  {
+    globalIndex const * cols = patternDiag.getColumns( localRow ).dataIfContiguous();
+    pattern.insertNonZeros( localRow, cols, cols + patternDiag.numNonZeros( localRow ) );
+  }
+
+  // Add the nonzeros from coupling
+  this->addCouplingSparsityPattern( domain, dofManager, pattern.toView() );
+
+  // Finally, steal the pattern into a CRS matrix
+  localMatrix.assimilate< parallelDevicePolicy<> >( std::move( pattern ) );
+  localMatrix.setName( this->getName() + "/localMatrix" );
+
+  rhs.setName( this->getName() + "/rhs" );
+  rhs.create( dofManager.numLocalDofs(), MPI_COMM_GEOS );
+
+  solution.setName( this->getName() + "/solution" );
+  solution.create( dofManager.numLocalDofs(), MPI_COMM_GEOS );
+}
+
+void SolidMechanicsLagrangeContactBubbleStab::implicitStepSetup( real64 const & time_n,
+                                                                 real64 const & dt,
+                                                                 DomainPartition & domain )
+{
+
+  SolidMechanicsLagrangianFEM::implicitStepSetup( time_n, dt, domain );
+
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&] ( string const &,
```

Comment:
why doing this every time step?

Reply:
yeah, if theory we don't need to. I mean... we only need to do it for newly created elements. I guess I could move this somewhere.

Reply:
perhaps there can be a specific function that updates newly created elements.